haystack/components/generators/hugging_face_api.py

```diff
@@ -208,8 +208,8 @@ def _run_non_streaming(self, prompt: str, generation_kwargs: Dict[str, Any]):
         meta = [
             {
                 "model": self._client.model,
-                "finish_reason": tgr.details.finish_reason,
-                "usage": {"completion_tokens": len(tgr.details.tokens)},
+                "finish_reason": tgr.details.finish_reason if tgr.details else None,
```

Comment:
@silvanocerza this was failing in local tests because details is now optional field. I added this guard here, in chat version and TGI equivalents. Double check that guarding logic please